src/index.js

```diff
@@ -11,17 +11,10 @@ export function WebpackPnpExternals(options={}) {
         pnpApi = null;
     }
     return function() {
-        const arg1 = arguments[0];
-        const arg2 = arguments[1];
-        const arg3 = arguments[2];
-        let context = arg1;
-        let request = arg2;
-        let callback = arg3;
-        // in case of webpack 5
-        if (arg1 && arg1.context && arg1.request) {
-            context = arg1.context;
-            request = arg1.request;
-            callback = arg2;
+        let [{ context, request }, callback] = arguments;
```

Comment:
We actually need the reverse of this (assume 4, fallback to 5) instead of what you have (assume 5, fallback to 4). As you have implemented here, it will throw an error on webpack 4 when it tries to destructure props out of the first arg (context) which is a string.

Also, I'm not sure what you mean about arguments getting overridden by parameters. The arguments object should always retain the original arguments regardless of what we do with the argument variables we've defined in the signature. What specifically is the weird behavior you're seeing?

Reply:
That's fine, I can reverse it.

After looking into it more, it looks like webpack is internally checking the function parameter length and calling it with the deprecated signature, which gives the warning I was originally trying to avoid.

In order to get around that, we need to destructure `arguments` directly rather than rely on the function parameters.

**From [webpack/lib/ExternalModuleFactoryPlugin.js](https://github.com/webpack/webpack/blob/8a3e79b42a81180909162ea1e9a5f8ae1991e0f0/lib/ExternalModuleFactoryPlugin.js):**
```
if (externals.length === 3) {
  // TODO webpack 6 remove this
  callDeprecatedExternals(
    externals,
    context,
    dependency.request,
    cb
  );
} else {
  externals(
    {
      context,
      request: dependency.request
    },
    cb
  );
}
```